discussment-learning/src/main/java/org/danekja/discussment/learning/panel/article/ArticleTextPanel.java

```diff
@@ -0,0 +1,55 @@
+package org.danekja.discussment.learning.panel.article;
+
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.panel.Panel;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+import org.apache.wicket.model.PropertyModel;
+import org.danekja.discussment.core.domain.Discussion;
+import org.danekja.discussment.core.domain.Post;
+import org.danekja.discussment.core.service.PostService;
+import org.danekja.discussment.learning.domain.Article;
+import org.danekja.discussment.ui.wicket.panel.discussion.DiscussionPanel;
+
+/**
+ * The class creates the panel which contains the specific article with its discussion. Can be added to a separate page.
+ *
+ * Date: 30.1.18
+ *
+ * @author Jiri Kryda
+ */
+public class ArticleTextPanel extends Panel {
+
+    private IModel<Article> articleModel;
+    private IModel<Discussion> discussionModel;
+
+    private PostService postService;
+
+
+    /**
+     * Constructor for creating the panel which contains the specific article and its discussion.
+     *
+     * @param id id of the element into which the panel is inserted
+     * @param article article in the panel
+     * @param discussion discussion in the panel
+     * @param postService instance of the post service
+     */
+    public ArticleTextPanel(String id, IModel<Article> article, IModel<Discussion> discussion, PostService postService){
```

Comment:
Only Article passed as parameter, no discussion. Discussion model can be created internaly using article model inside of this component.